### PR TITLE
Add collection typedef

### PR DESCRIPTION
**Pull request by agnat**
In preparation of implementing the binary output format `application/vnd.google.protobuf` I'd like to land this little clean-up.

### CODE REVIEW

File: prometheus/registry.hh [hunk 1]
```diff
@@ -23,10 +21,12 @@ namespace prometheus {
       CollectorRegistry();
       ~CollectorRegistry();
 
+      typedef Collector::collection_type collection_type;
```
Comment (korfuri) [inline prometheus/registry.hh]: Is there a specific reason you're re-typedefing here instead of using `using` statements?

**Review by korfuri — CHANGES_REQUESTED** [review]
Look good overall, thanks!
Please rebase as right now there are some merge conflicts :)

File: prometheus/registry.hh (hunk 1 above)
Comment (agnat) [reply prometheus/registry.hh]: Old habits die hard. Will update.

**Comment by agnat** [comment]
Done.

**Comment by agnat** [comment]
Depends on #14.

**Review by korfuri — APPROVED** [review]

**Pull request merged**